api/src/main/java/org/openmrs/module/appointmentscheduling/api/AppointmentService.java

```diff
@@ -789,8 +786,8 @@ AppointmentStatusHistory saveAppointmentStatusHistory(
 	 */
     @Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
 	List<Appointment> getAppointmentsByConstraints(Date fromDate, Date toDate,
-			Location location, Provider provider, AppointmentType type,
-			AppointmentStatus status) throws APIException;
+												   Location location, Provider provider, AppointmentType type,
+												   AppointmentStatus status, VisitType visitType, Visit visit) throws APIException;
```

Comment:
1) I'll fix the formatting issues
2) I think restricting one appointment per visit makes sense  hence getAppointmentByVisit should  suffice. My challenge was how to call the method once the visit param is passed in the url. However I think I can achieve this in the doSearch override method in which I should check cases when all other params are null but visit param is provided and call the getAppointmentByVisit method, is this the right way of doing it?
3. On editing existing API methods, I'll a thread on this on openmrs talk to get to understand this more. I'll then update the pr.

Reply:
recreated the branch to remove the formatting issues, with this change to access the pr, first go to the commits tab.

Reply:
Ah, okay... I hadn't been thinking or looking at the REST search handler, so I understand why you did it the way you did now...  this actually makes me change my mind, I think, and I actually think you can keep it the way you have it... although it doesn't really make sense to search for an appointment based on visit *and* other parameters, it does seem easier to just provide a single REST endpoint that handles everything... sorry about that, hopefully you haven't refactored too much, but I think you can keep it as-is, with the exception that you should create new service methods in AppointmentService instead of overriding the existing ones.  Does that make sense?

Let me know when you've got the formatting cleaned up, and I'll take a closer look!

Reply:
That makes sense, I'll update you as soon as I update the pr

Reply:
Fixed most of the white spaces issues. Also updated the pr with regards to your reviews. Kindly review and give feedback